src/chapters/orm-relationships/many-to-many.rst

```diff
@@ -0,0 +1,332 @@
+
+Exactly *how* this relationship is stored utilizes a new type of SQL table.
+
+Join Tables
+^^^^^^^^^^^
+
+Think about how relationships are established at the database level. One-to-one and one-to-many relationships are facilitated by the use of a foreign key column on one side of the relationship. Our ``event`` table has two foreign key columns: ``event_category_id`` and ``event_details_id``. 
+
+For a given row in ``event``, the column ``event_category_id`` contains the primary key of the row in ``event_category`` that the ``event`` row is related to, an similarly for ``event_details_id``. 
+
+The only difference is the *number* of different ``event`` rows that may have the same value of ``event_category_id`` and ``event_details_id``. The ``event``/``event_category`` relationship is many-to-one, so *many* event rows may have the same ``event_category_id`` value. The ``event``/``event_details`` relationship is one-to-one, so *only one* event row may have a given value in ``event_details_id``. 
+
+.. index:: ! join table
+
+Using foreign and primary keys to create many-to-many relationships is a bit trickier. In order to relate rows in ``event`` to rows in ``tag`` we need need a third table, known as a **join table**. A join table consists of two columns, each of which is a foreign key column to another table. Each row in a join table represents a relationship between one row of each of the two tables. This technique enables many-to-many relationships.
+
+Consider some example data in our ``event`` and ``tag`` tables.
+
+.. list-table:: Sample ``event`` data
+   :header-rows: 1
+
+   * - id
+     - name
+     - event_category_id
+     - event_details_id
+   * - 13
+     - WWDC
+     - 2
+     - 14
+   * - 15
+     - SpringOne Platform
+     - 2
+     - 16
+   * - 17
+     - Java meetup
+     - 3
+     - 18
+   
+.. list-table:: Sample ``event_category`` data
+   :header-rows: 1
+
+   * - id
+     - name
+   * - 2
+     - Conference
+   * - 3
+     - Meetup
+
+.. list-table:: Sample ``tag`` data
+   :header-rows: 1
+
+   * - id
+     - name
+   * - 4
+     - ios
+   * - 5
+     - spring
+   * - 6
+     - java
+
+A join table for these two tables would be called ``event_tags``, and would have two columns, ``event_id`` and ``tag_id``. Each of these columns are foreign key columns into their respective tables. 
+
+If we want to relate the ``ios`` tag to the ``WWDC`` event, we create a new row in ``event_tags``:
+
+.. list-table:: A join table with a single relationship
+   :header-rows: 1
+
+   * - events_id
+     - tags_id
+   * - 13
+     - 4
+
+We can do this again and again to generate more relationships. Let's revisit the :ref:`many-to-many diagram <many-to-many-figure>` from earlier in the chapter. 
```

Comment:
if it is the same figure, it seems counter-productive to link back to it. I'd remove the ref or change the ref to the section heading that introduces the many to many relationship in the chapter